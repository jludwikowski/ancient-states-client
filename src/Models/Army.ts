import {Entity} from './Entity';
import Leader from './Leader';
import UnitReference from "./UnitReference";

export default interface Army extends Entity {
    commander: Leader | null,
    units: UnitReference[]
    owner: number,
    position: string,
    status: string,
    orderDate?: string | null,
    targetId?: number | null,
}
