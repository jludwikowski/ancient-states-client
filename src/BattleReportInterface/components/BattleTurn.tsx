import React from "react";
import Army from "../../Models/Army";
import {ArmyTurnReport} from "./ArmyTurnReport";

interface Props {
    armiesList: Army[]
}

export const BattleTurn: React.FunctionComponent<Props> = props => (
    <li className="list-group-item">
        <ul className="list-group">
            {props.armiesList.map((army) => (
                <React.Fragment key={army.id}>
                    <ArmyTurnReport army={army} />
                </React.Fragment>
            ))}
        </ul>
    </li>
)
