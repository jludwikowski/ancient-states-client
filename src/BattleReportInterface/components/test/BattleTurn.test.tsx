import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Army from '../../../Models/Army';
import { BattleTurn } from '../BattleTurn';

describe('BattleTurn', () => {

    const armiesList:Army[] = [{
        id: 7,
        name: 'New Army',
        description: '',
        status: 'camp',
        orderDate: null,
        targetId: null,
        position: '0,0',
        owner: 1,
        commander: null,
        units: [
            {
                number: 9802,
                level: 1,
                baseUnit: 1,
            }
        ]
    },{
        id: 1,
        name: 'New Army',
        description: '',
        status: 'camp',
        orderDate: null,
        targetId: null,
        position: '0,0',
        owner: 2,
        commander: null,
        units: [
            {
                number: 300,
                level: 1,
                baseUnit: 1,
            }
        ]
    },
    ];

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BattleTurn armiesList={armiesList} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<BattleTurn armiesList={armiesList} />);
        expect(component).toMatchSnapshot();
    });
});
