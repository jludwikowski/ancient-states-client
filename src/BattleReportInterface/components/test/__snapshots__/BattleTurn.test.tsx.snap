// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BattleTurn Should match snapshot 1`] = `
<li
  className="list-group-item"
>
  <ul
    className="list-group"
  >
    <ArmyTurnReport
      army={
        Object {
          "commander": null,
          "description": "",
          "id": 7,
          "name": "New Army",
          "orderDate": null,
          "owner": 1,
          "position": "0,0",
          "status": "camp",
          "targetId": null,
          "units": Array [
            Object {
              "baseUnit": 1,
              "level": 1,
              "number": 9802,
            },
          ],
        }
      }
    />
    <ArmyTurnReport
      army={
        Object {
          "commander": null,
          "description": "",
          "id": 1,
          "name": "New Army",
          "orderDate": null,
          "owner": 2,
          "position": "0,0",
          "status": "camp",
          "targetId": null,
          "units": Array [
            Object {
              "baseUnit": 1,
              "level": 1,
              "number": 300,
            },
          ],
        }
      }
    />
  </ul>
</li>
`;
